### PR TITLE
More hg usecases tweaks

### PR DESCRIPTION
**Pull request by odcinek**

### CODE REVIEW

File: lib/between_meals/repo/hg.rb [hunk 1]
```diff
@@ -115,9 +115,17 @@ def head_parents
       end
 
       def last_author
-        { :email => template('author').match(/^(.*)<(.*)>$/)[2] }
-      rescue
-        { :email => nil }
+        # rubocop:disable Lint/HandleExceptions
+        begin
+          return { :email => template('author').match(/^(.*)<(.*)>$/)[2] }
+        rescue
```
Comment (jaymzh) [inline lib/between_meals/repo/hg.rb]: why not rescue only the Match exception here - you know exactly what you're looking for - and then you wouldn't need the rubocop exceptions

File: lib/between_meals/repo/hg.rb [hunk 2]
```diff
@@ -144,6 +152,17 @@ def name
...
 
+      def status
+        s = Mixlib::ShellOut.new(
+          "#{@bin} status",
+          :cwd => File.expand_path(@repo_path)
+        ).run_command
+        s.error!
+        s.stdout
+      rescue
+        nil
```
Comment (jaymzh) [inline lib/between_meals/repo/hg.rb]: don't use `error!` if you plan to catch the exception... what's the point?

File: lib/between_meals/repo/hg.rb (hunk 1 above)
Comment (odcinek) [inline lib/between_meals/repo/hg.rb]: That rule is about the rescue block being empty, which is fine here, not about catching specific exception classes

File: lib/between_meals/repo/hg.rb (hunk 2 above)
Comment (odcinek) [inline lib/between_meals/repo/hg.rb]: The point is returning with nil in case of non-zero code, instead of whatever happens to be in s.stdout

Comment (jaymzh) [inline lib/between_meals/repo/hg.rb]: return nil if s.error

File: lib/between_meals/repo/hg.rb (hunk 1 above)
Comment (jaymzh) [inline lib/between_meals/repo/hg.rb]: consider:

```
{
  /(.*)(.*)/ => 2,
  /(.*)/ => 1.
}.each do |re, in|
  m = str.match(re)
  return m[in] if m
end

  
```

**Pull request merged**